pkg/cmd/roachtest/backup.go

```diff
@@ -262,4 +275,163 @@ func registerBackup(r *testRegistry) {
 			m.Wait()
 		},
 	})
+
+	r.Add(testSpec{
+		Name:       "backupKMS",
+		Owner:      OwnerBulkIO,
+		Cluster:    makeClusterSpec(3),
+		MinVersion: "v20.2.0",
+		Run: func(ctx context.Context, t *test, c *cluster) {
+			if cloud == gce {
+				t.Skip("backupKMS roachtest is only configured to run on AWS", "")
+			}
+
+			// ~10GiB - which is 30Gib replicated.
+			rows := 976562
+			dest := c.name
+
+			if local {
+				rows = 100
+				dest += fmt.Sprintf("%d", timeutil.Now().UnixNano())
+			}
+
+			c.Put(ctx, workload, "./workload")
+			c.Put(ctx, cockroach, "./cockroach")
+
+			// NB: starting the cluster creates the logs dir as a side effect,
+			// needed below.
+			c.Start(ctx, t)
+			c.Run(ctx, c.All(), `./workload csv-server --port=8081 &> logs/workload-csv-server.log < /dev/null &`)
+			time.Sleep(time.Second) // wait for csv server to open listener
+
+			c.Run(ctx, c.Node(1), "./workload", "fixtures", "import", "bank",
+				"--db=bank", "--payload-bytes=10240", "--ranges=0", "--csv-server", "http://localhost:8081",
+				fmt.Sprintf("--rows=%d", rows), "--seed=1", "{pgurl:1}")
+
+			conn := c.Conn(ctx, 1)
+			m := newMonitor(ctx, c)
+			m.Go(func(ctx context.Context) error {
+				_, err := conn.ExecContext(ctx, `
+					CREATE DATABASE restoreA;
+					CREATE DATABASE restoreB;
+				`)
+				return err
+			})
+			m.Wait()
+
+			var kmsURIA, kmsURIB string
+			var err error
+			m = newMonitor(ctx, c)
+			m.Go(func(ctx context.Context) error {
+				t.Status(`running encrypted backup`)
+				kmsURIA, err = getAWSKMSURI(KMSRegionAEnvVar, KMSKeyARNAEnvVar)
+				if err != nil {
+					return err
+				}
+
+				kmsURIB, err = getAWSKMSURI(KMSRegionBEnvVar, KMSKeyARNBEnvVar)
+				if err != nil {
+					return err
+				}
+
+				kmsOptions := fmt.Sprintf("KMS=('%s', '%s')", kmsURIA, kmsURIB)
+				_, err := conn.ExecContext(ctx,
+					`BACKUP bank.bank TO 'nodelocal://1/kmsbackup/`+dest+`' WITH `+kmsOptions)
+				return err
+			})
+			m.Wait()
+
+			// Restore the encrypted BACKUP using each of KMS URI A and B separately.
+			m = newMonitor(ctx, c)
+			m.Go(func(ctx context.Context) error {
+				t.Status(`restore using KMURIA`)
+				if _, err := conn.ExecContext(ctx,
+					`RESTORE bank.bank FROM $1 WITH into_db=restoreA, kms=$2`,
+					`nodelocal://1/kmsbackup/`+dest, kmsURIA,
+				); err != nil {
+					return err
+				}
+
+				t.Status(`restore using KMURIB`)
+				if _, err := conn.ExecContext(ctx,
+					`RESTORE bank.bank FROM $1 WITH into_db=restoreB, kms=$2`,
+					`nodelocal://1/kmsbackup/`+dest, kmsURIB,
+				); err != nil {
+					return err
+				}
+
+				t.Status(`fingerprint`)
+				fingerprint := func(db string) (string, error) {
```

Comment:
Not needed to be addressed in this PR, but I think we've defined this at least one other place and I think it would be generally useful to have a `fingerprint` utility for backup/restore roachtests to use rather than have each test define their own.